sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -84,119 +84,77 @@ export class MessageSender extends LinkEntity<AwaitableSender> {
     this._retryOptions = retryOptions;
     this._onAmqpError = (context: EventContext) => {
       const senderError = context.sender && context.sender.error;
-      if (senderError) {
-        const err = translate(senderError);
-        log.error(
-          "[%s] An error occurred for sender '%s': %O.",
-          this._context.connectionId,
-          this.name,
-          err
-        );
-      }
+      log.error(
+        "[%s] 'sender_error' event occurred on the sender '%s' with address '%s'. " +
+          "The associated error is: %O",
+        this._context.connectionId,
+        this.name,
+        this.address,
+        senderError
+      );
+      // TODO: Consider rejecting promise in trySendBatch() or createBatch()
```

Comment:
This seems like a good idea.  

Perhaps we could just do something similar to what we do in Batching Receiver where each operation essentially registers itself with the error related events.

(It could be a logical deregister/register where we just assign to a field rather than doing an on(event) assignment each time.)

Reply:
Yeah, logged an issue long ago #8656